megamek/src/megamek/client/bot/princess/FireControl.java

```diff
@@ -1892,8 +1894,11 @@ private FiringPlan getDiveBombPlan(final Entity shooter,
         }
 
         // not having any bombs (due to expenditure/damage)
-        if (shooter.getBombs(BombType.F_GROUND_BOMB).isEmpty()) {
+        List<BombMounted> groundBombs = shooter.getBombs(BombType.F_GROUND_BOMB);
+        if (groundBombs.isEmpty()) {
             return diveBombPlan;
+        } else {
```

Comment:
This else block isn't required since the function returns if `groundBomb.isEmpty()` is true.

Reply:
Sorry, just force of habit.  I'll try to keep it in check.